pkg/microservice/aslan/core/system/service/registry.go

```diff
@@ -199,36 +206,100 @@ func ListAllRepos(log *zap.SugaredLogger) ([]*RepoInfo, error) {
 	return repoInfos, nil
 }
 
+type newTags struct {
+	Resp []*RepoImgResp
+	m    *sync.Mutex
+}
+
 func ListReposTags(registryInfo *commonmodels.RegistryNamespace, names []string, logger *zap.SugaredLogger) ([]*RepoImgResp, error) {
 	var regService registry.Service
 	if registryInfo.AdvancedSetting != nil {
 		regService = registry.NewV2Service(registryInfo.RegProvider, registryInfo.AdvancedSetting.TLSEnabled, registryInfo.AdvancedSetting.TLSCert)
 	} else {
 		regService = registry.NewV2Service(registryInfo.RegProvider, true, "")
 	}
+
+	endPoint := registry.Endpoint{
+		Addr:      registryInfo.RegAddr,
+		Ak:        registryInfo.AccessKey,
+		Sk:        registryInfo.SecretKey,
+		Namespace: registryInfo.Namespace,
+		Region:    registryInfo.Region,
+	}
 	repos, err := regService.ListRepoImages(registry.ListRepoImagesOption{
-		Endpoint: registry.Endpoint{
-			Addr:      registryInfo.RegAddr,
-			Ak:        registryInfo.AccessKey,
-			Sk:        registryInfo.SecretKey,
-			Namespace: registryInfo.Namespace,
-			Region:    registryInfo.Region,
-		},
-		Repos: names,
+		Endpoint: endPoint,
+		Repos:    names,
 	}, logger)
 
 	images := make([]*RepoImgResp, 0)
 	if err == nil {
 		for _, repo := range repos.Repos {
+			// find all tags from image tags db
+			imageTags, err := commonrepo.NewImageTagsCollColl().Find(&commonrepo.ImageTagsFindOption{
+				RegistryID:  registryInfo.ID.Hex(),
+				RegProvider: registryInfo.RegProvider,
+				ImageName:   repo.Name,
+				Namespace:   registryInfo.Namespace,
+			})
+			dbTags := make(map[string]*commonmodels.ImageTag, 0)
+			if err != nil {
+				if err != mongo.ErrNoDocuments && err != mongo.ErrNilDocument {
+					logger.Errorf("find tags of the image %s from db error: %v", repo.Name, err)
+					return nil, err
```

Comment:
whe return error here since the following inserting logic won't block the entire progress?